src/utils/hooks/use-animation.ts

```diff
@@ -0,0 +1,53 @@
+import { useRef, useEffect } from 'react'
+import { Animated } from 'react-native'
+
+export const useAnimatedValue = (initialValue: number): Animated.Value => {
+  const ref = useRef(new Animated.Value(initialValue))
+  return ref.current
+}
+
+interface BaseAnimationConfig {
+  initialValue?: number
+}
+
+export type TimingAnimationConfig = BaseAnimationConfig &
+  ({ type: 'timing' } & Animated.TimingAnimationConfig)
+
+export type SpringAnimationConfig = BaseAnimationConfig &
+  ({ type: 'spring' } & Animated.SpringAnimationConfig)
+
+export type UseAnimationConfig = TimingAnimationConfig | SpringAnimationConfig
+
+const getInitialValue = (config: UseAnimationConfig): number => {
+  if (typeof config.initialValue !== 'undefined') {
+    return config.initialValue
+  } else {
+    return config.toValue as number // TODO deal with other types possibilities here
+  }
+}
+
+/**
+ * Returns an `Animated.Value` object initialized with the specified config.
+ * Its initial value is either `initialValue` or if this is not provided, initial
+ * `toValue` is used.
+ *
+ * When `toValue` value changes, an animation will run with the specified config.
+ */
+export const useAnimation = (config: UseAnimationConfig): Animated.Value => {
+  const animatedValue = useAnimatedValue(getInitialValue(config))
```

Comment:
Is there a plus value of having a dedicated hook for `useAnimatedValue`, since it's only a `ref`?

Reply:
Not really, it just makes usage simpler and prevents having multiple imports.

```tsx
// Previous usage
import { useRef } from 'react'
import { Animated } from 'react-native'

const anim = useRef(new Animated.Value(initialValue)).current
// or 
const [anim] = useState(new Animated.Value(initialValue))

// New Usage
import { useAnimatedValue } from '../../utils/hooks/use-animation'

const anim = useAnimatedValue(initialValue)
```

It's really syntactic sugar... Personally, I always forget the `.current` after `useRef` :stuck_out_tongue_closed_eyes: . However, it adds an additional hook which could be prevented... Should I remove it?

Reply:
I'd lean to keep short (and standard) code in the body to avoid having to look around :)

Reply:
I agree that it would add another place to lookup...

Updated with latest version of `develop` and removed the `useAnimatedValue` hook :wink: